server/test/auth/ApiKeyGrantsTest.java

```diff
@@ -0,0 +1,66 @@
+package auth;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class ApiKeyGrantsTest {
+
+  private static final String PROGRAM_1_SLUG = "program-1";
+  private static final String PROGRAM_2_SLUG = "program-2";
+  private ApiKeyGrants subject;
+
+  @Before
+  public void setUp() {
+    subject = new ApiKeyGrants();
+  }
+
+  @Test
```

Comment:
Add the following tests
- No setup, permissions aren't allowed (IE an unknown Slug isn't allowed)
- Multiple SLUGS (One read, one write perhaps) have appropriate permissions.

Reply:
> No setup, permissions aren't allowed (IE an unknown Slug isn't allowed)

done

> Multiple SLUGS (One read, one write perhaps) have appropriate permissions. (IE we don't mix up permissions.)

ah sharp eye, done.